week1/homework/src/server.js

```diff
@@ -1,17 +1,40 @@
 'use strict';
 
+const url = require('url');
 const http = require('http');
+// const fs = require('fs');
 
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
 function createServer(port) {
   let state = 10;
+  let newState = state;
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
-  });
+    const requestedUrl = url.parse(request.url, true);
 
+    if (requestedUrl.pathname === '/state') {
+      response.writeHead(200, { 'Content-type': 'application/json' });
+      response.end(JSON.stringify({ state: newState }));
+    } else if (requestedUrl.pathname === '/add') {
+      response.writeHead(200, { 'Content-type': 'application/json' });
+      newState = newState + 1;
```

Comment:
maybe you already know and just prefer it this way, you can make this shorter by doing `newState++` to increment a variable and `newState--` to decrement :)

Reply:
I changed.